libbpf-rs/src/ringbuf.rs

```diff
@@ -0,0 +1,146 @@
+use core::ffi::c_void;
+use std::boxed::Box;
+use std::ptr;
+use std::slice;
+use std::time::Duration;
+
+use crate::*;
+
+/// The canonical interface for managing a collection of [`RingBuffer`]s.
+///
+/// `ringbuf`s are a special kind of [`Map`], used to transfer data between
+/// [`Program`]s and userspace.  As of Linux 5.8, the `ringbuf` map is now
+/// preferred over the `perf buffer`.
+pub struct RingBufferManager<F>
+where
+    F: FnMut(&[u8]) -> i32 + 'static,
+{
+    ptr: *mut libbpf_sys::ring_buffer,
+    ringbufs: Vec<Box<F>>,
+}
+
+impl<F> RingBufferManager<F>
+where
+    F: FnMut(&[u8]) -> i32 + 'static,
+{
+    /// Add a new ringbuf `map` and associated `callback` to this ring buffer
+    /// manager. The callback should take one argument, a slice of raw bytes,
+    /// and return an i32.
+    ///
+    /// Non-zero return values will stop ring buffer consumption early.
+    ///
+    /// The callback provides a raw byte slice. You may find libraries such as
+    /// [`plain`](https://crates.io/crates/plain) helpful.
+    pub fn add_ringbuf(&mut self, map: &Map, callback: F) -> Result<&mut Self> {
```

Comment:
can we use just `add()` name to stay in line with libbpf's `ring_buffer__add()` API?

Reply:
Sure